src/components/RegistrationModal.tsx

```diff
@@ -61,158 +53,37 @@ const RegistrationModal = ({
-        )}
-        <Button
-          type="primary"
-          htmlType="submit"
-          className="RegistrationModal__footerBtn"
-        >
-          Choose handle
-        </Button>
-      </div>
-    </Form>
-  );
-
-  /**
-   * Select Existing Handle UI
-   */
-  const selectHandleContent = (
-    <div>
-      <p>You have multiple handles associated with your account.</p>
-      <h3>Select an account:</h3>
-      <div className="RegistrationModal__registrations">
-        {registrations.map((registration: Registration) => (
-          <button
-            className={`RegistrationModal__registration${
-              registration === selectedRegistration
-                ? " RegistrationModal__registration--selected"
-                : ""
-            }`}
-            onClick={() => setRegistration(registration)}
-            key={registration.dsnpUserURI}
-          >
-            <UserAvatar
-              icon={iconForRegistration(registration)}
-              profileAddress={registration.dsnpUserURI}
-              avatarSize="small"
-            />
-            {`@${registration.handle}`}
-          </button>
-        ))}
-      </div>
-      <div className="RegistrationModal__footer">
-        <button
-          onClick={() => setIsCreatingRegistration(true)}
-          className="RegisrationModal__changeMode"
-        >
-          Create New Handle
-        </button>
-        <Button
-          className="RegistrationModal__footerBtn"
-          key="post"
-          type="primary"
-          disabled={!selectedRegistration}
-          onClick={commitRegistration}
-        >
-          Select Handle
-        </Button>
-      </div>
-    </div>
-  );
-
   /**
    * Create a popover with either registration or handle selection UI.
    */
   return (
     <Popover
       placement="bottomRight"
       visible={visible}
+      trigger="click"
+      onVisibleChange={() => setRegistrationVisible(!visible)}
       content={
-        <div className="RegistrationModal">
-          <Button
-            type="link"
-            className="RegistrationModal__cancel"
-            onClick={onCancel}
-          >
-            Cancel
-          </Button>
-          <h2>Welcome!</h2>
-          {isCreatingRegistration ? registerNewHandleForm : selectHandleContent}
-        </div>
+        <EditRegistration
+          logout={logout}
+          isCreatingRegistration={isCreatingRegistration}
+          walletAddress={walletAddress}
+          onIdResolved={onIdResolved}
+          registrations={registrations}
+        />
       }
     >
-      {children}
+      <div className="RegistrationHub__userBlock">
+        <UserAvatar
+          icon={profile?.icon?.[0]?.href}
+          profileAddress={userId}
+          avatarSize="small"
+        />
+        <div className="RegistrationHub__userTitle">
+          {profile?.handle
+            ? "@" + profile.handle
+            : profile?.name || profile?.fromId}
+        </div>
+      </div>
```

Comment:
Should this be its own component or are we happy with it like this?

Reply:
I think you should go back to rendering `{children}` here and pull this code up to the calling context. `RegistrationModal` shouldn't have responsibility for rendering the button that triggers it. Unfortunately, it does need to wrap the button so it can position itself correctly.